src/libutil/strutil.cpp

```diff
@@ -330,15 +330,24 @@ Strutil::wordwrap (string_view src, int columns, int prefix)
 
 
 
-namespace {
-static std::locale loc = std::locale::classic();
+
+static std::locale& loc ()
+{
+    // Rather than have a file-level static locale which could suffer from
+    // the static initialization order fiasco, make it a static pointer
+    // inside the function that is allocated the first time it's needed. It
+    // will "leak" in the sense that this one locale is never deleted, but
+    // so what?
+    static std::locale *loc = new std::locale (std::locale::classic());
```

Comment:
Why not simply

    static std::locale loc = std::locale::classic();

?

Moreover, one needs to be aware that this is not thread safe.

Reply:
If it's static rather than a pointer, you can have the same order fiasco upon destruction. This one "leaks" (never frees) the loc, but it will definitely not free it before some other destructor might need it (say, if a destructor for some reason used any of the strutil functions).

Initialization of static variables within functions *is* thread-safe, that's guaranteed by C++11. 

You can see the discussion of this solution here: https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use

Reply:
> If it's static rather than a pointer, you can have the same order fiasco upon destruction.

Ok, hadn't thought about destruction.

> Initialization of static variables within functions is thread-safe, that's guaranteed by C++11

Ok, that's new with C++11 then, I didn't know.

Reply:
Or maybe it was thread-safe all along? https://stackoverflow.com/a/8102145/393756

Reply:
Sorry for the spam, but there are more details here: https://stackoverflow.com/a/1661564/393756

In particular, it looks like it wasn't thread-safe before Visual Studio 2015...

Reply:
So all in all, while I agree with the fix on principle, I just want to reiterate that it probably won't be thread-safe with VS 2012 and VS 2013 (both used quite a lot because they are the mandatory compilers for 3ds Max 2015/2016, and probably some versions of Maya as well).

Note that it's necessary a big deal...

Reply:
Note that it's _not_ necessary a big deal...

Reply:
I'll put in a fix for that...